packages/babel-plugin-minify-mangle-names/src/index.js

```diff
@@ -103,26 +107,37 @@ module.exports = ({ types: t }) => {
           //   i = 0;
           // }
 
-          Object
-            .keys(scope.getAllBindings())
-            .filter((b) => {
-              const binding = scope.getBinding(b);
-
-              return scope.hasOwnBinding(b)
-                && !binding.path.isLabeledStatement()
-                && !mangler.isBlacklist(b, mangler.blacklist)
-                && (mangler.keepFnames ? !isFunction(binding.path) : true);
-            })
-            .map((b) => {
-              let next;
-              do {
-                next = getNext();
-              } while (!t.isValidIdentifier(next) || scope.hasBinding(next) || scope.hasGlobal(next) || scope.hasReference(next));
-              // TODO:
-              // re-enable this
-              // resetNext();
-              mangler.rename(scope, b, next);
-            });
+          const bindings = scope.getAllBindings();
+          const names = Object.keys(bindings);
+
+          for (let i = 0; i < names.length; i++) {
+            const oldName = names[i];
+            const binding = bindings[oldName];
+
+            if (
+              !scope.hasOwnBinding(oldName)
+              || binding.path.isLabeledStatement()
+              || mangler.isBlacklist(oldName)
+              || (mangler.keepFnames ? isFunction(binding.path) : false)
+            ) {
+              continue;
+            }
+
+            let next;
+            do {
+              next = getNext();
+            } while (
+              !t.isValidIdentifier(next)
+              || hop.call(bindings, next)
+              || scope.hasGlobal(next)
```

Comment:
I'm not sure. `scope is created (globals calculated from bindings) -> visitor applied` is what I thought. I'll check though.